cpmpy/expressions/core.py

```diff
@@ -542,7 +542,7 @@ def value(self):
         if any(a is None for a in arg_vals): return None
         # non-boolean
         elif self.name == "sum": return sum(arg_vals)
-        elif self.name == "wsum": return int(sum(arg_vals[0]*np.array(arg_vals[1])))
+        elif self.name == "wsum": return np.dot(arg_vals[0], arg_vals[1]).item()
```

Comment:
so this will return a float (even if it equals an int exactly), but in the solver code you do convert to an int in those cases, just a small inconsistency that I don't know what to do with.

Reply:
Aha, yes ok dan we need to convert it here too.
So will add a "round" check as I did in the objective translation. The "item" is apparently a good way of getting the Python value instead of the numpy value